sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -562,7 +562,7 @@ abstract class HadoopFsRelation private[sql](maybePartitionSpec: Option[Partitio
     })
   }
 
-  private[sql] def buildScan(
+  final private[sql] def buildScan(
```

Comment:
This method was marked as `final` at the very beginning, and wasn't designed to be overridden by subclasses. The `final` decorator was removed in #8035 to work around the refreshing issue.